pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go

```diff
@@ -326,6 +327,10 @@ func (dswp *desiredStateOfWorldPopulator) processPodVolumes(
 			allVolumesAdded = false
 			continue
 		}
+		uniqueVolumeName, err := dswp.desiredStateOfWorld.GetUniqueVolumeName(uniquePodName, volumeSpec)
+		if err == nil && !dswp.desiredStateOfWorld.PodExistsInVolume(uniquePodName, uniqueVolumeName) {
```

Comment:
if err != nil here, allVolumesAddedPreviously will still be true. I don't think that's correct.